File: solutions/erlang/raindrops/1/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(_Number) ->
    Sounds =
        lists:concat(
          [
           if ( _Number rem 3 =:= 0 ) -> "Pling"; true -> "" end,
           if ( _Number rem 5 =:= 0 ) -> "Plang"; true -> "" end,
           if ( _Number rem 7 =:= 0 ) -> "Plong"; true -> "" end
          ]
    ),
    if ( Sounds =:= "" ) -> integer_to_list(_Number); true -> Sounds end
.

File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    std::string convert(int number) {
        std::string sounds = "";
        if ( number % 3 == 0  ) sounds += "Pling";
        if ( number % 5 == 0  ) sounds += "Plang";
        if ( number % 7 == 0  ) sounds += "Plong";
        if ( sounds     == "" ) sounds = std::to_string(number);
        return sounds;
    }
}  // namespace raindrops
